bundle/src/main/java/com/adobe/acs/commons/componentcloner/impl/ComponentClonerServlet.java

```diff
@@ -0,0 +1,118 @@
+/*
```

Comment:
Overall this looks pretty solid, very nice!

I think the big thing I'd like to understand (and that you'll want to put on the official PR) is the use case for this feature - when/why would someone use this.  I think you've already told me, but capturing it in writing to help everyone (particularly the repo maintainers) understand will be helpful in getting this feature through the process.

Reply:
If you have a very complex component that has children/grandchildren/etc nodes, or just a lot of properties and want to retain a lot of those properties without having to re-author, you can do the following:

- Drop the component cloner on the page
- Navigate to the component/node you wish to clone
- Click save and that entire component/node structure will be cloned to where ever the original component cloner was placed on a parsys

This is very useful if your org's implementation of AEM is atomic and your components consist of 3, 4, or even 10+ micro components (atoms). Rather than the author having to configure those 3, 4, or 10+ atoms, then can clone a preexisting cluster and then just modify the verbiage. 

For example, imagine you have an existing atomic component-cluster that contains a heading atom, an RTE atom, a container atom, 6 news card atoms, and 2 CTA atoms. In total, that is 11 atom components that would all need to be dropped on the parsys and authored. Now imagine an author wants to reuse that same set up, but just change the text. They could then drop the component cloner where they wish for this cluster to reside and clone it over. They don't need to mess with reconfiguring all properties to mimic the structure since it was cloned over. They just need to change the text properties. 

I have a feeling this is very niche, but I wanted to get it out there to get feedback on it

Reply:
Fantastic - I think you'll want to add this information to the PR when you submit to ACS Commons.